src/crimson/osd/backfill_state.cc

```diff
@@ -162,7 +162,7 @@ void BackfillState::Enqueuing::maybe_update_range()
     peering_state().scan_log_after(primary_bi.version, func);
     logger().debug("{}: scanning projected log", __func__);
     pg().get_projected_log().scan_log_after(primary_bi.version, func);
-    primary_bi.version = pg().get_projected_last_update();
+    primary_bi.version = peering_state().get_pg_committed_to();
```

Comment:
We have just finished to scan the projected log here, the version should actually be the projected_last_update, right?
I think this case is different than the other version updates in this commit.

Reply:
Um, not really, as illustrated in https://github.com/ceph/ceph/pull/59615#discussion_r1825503095, scanning the log can't guarantee that the uncommitted object creation be included properly, so we need to make sure the version of the backfill interval progresses as conservatively as possible.